wordpress-mobile/WordPress-iOS
# Passkeys: Add support for "Log in with security keys"

## Description
- Adds support for "log in with security keys" by removing `applinks:*.wordpress.com` from associated domains

## Notes
- I cherry-picked https://github.com/wordpress-mobile/WordPress-iOS/pull/21739/commits/e62fbeeb50a7b488976dd2400f59bbb4bb0dca7c (Thanks @Ecarrion) 
- Removing `applinks:*.wordpress.com` from associated domains will disable deep linking into the Reader for `*.wordpress.com/:post_year/:post_month/:post_day/:post_name` links (ref: https://github.com/wordpress-mobile/WordPress-iOS/pull/16319)
- We've decided that removing deep linking support for `*.wordpress.com/:post_year/:post_month/:post_day/:post_name` links is an acceptable trade-off to get passkeys working (Slack ref: p1699395576443799/1698884760.893009-slack-C0180B5PRJ4)

## How to test
- Log into an account that has security keys enabled using your username and password
- In the 2FA step, select "Use a security key"
- Verify you're able to log in


https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/aa9ede60-5e0a-4a27-a98d-b4755cfef57e



## Regression Notes
1. Potential unintended areas of impact
Log in, Deep linking into Reader for `*.wordpress.com` links

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually for log in, reached a decision that removing support for `*.wordpress.com` deep linking was an acceptable trade-off (see notes above)

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:
